### PR TITLE
Rename the module and assumed file path.

### LINKED ISSUE #314: PSModulePath is not set up properly
Today the GoogleCloudPowerShell module gets installed into "${env:ProgramFilesx86}\Google\Cloud SDK\google-cloud-sdk\platform", and the installer adds the path to the GoogleCloudPowerShell folder to $env:PSModulePath.

This setup prevents the module from being loaded using Import-Module GoogleCloudPowerShell. You can load it using the full path to the module folder or using the full path to the psd1 file it contains, but you can't load it with Import-Module. This setup also prevents Get-Command -Module GoogleCloudPowerShell from working. Generally, anything that is dependent on lookup by module name (dependencies taken by other modules, for example) will not work.

To fix this, $env:PSModulePath should contain the folder _above_ the GoogleCloudPowerShell folder rather than the GoogleCloudPowerShell folder itself. Personally I recommend creating a PowerShell folder at the same level as GoogleCloudPowerShell, moving the GoogleCloudPowerShell folder into the PowerShell folder, and setting up the PowerShell folder you create as the item that you add to $env:PSModulePath.  That is the best solution to this issue, keeping PSModulePath folders clean and PowerShell focused so that you don't impact PowerShell performance, and setting your module up to be discoverable by name properly.

### PR DESCRIPTION
**Pull request by chrsmith**
Changes the module name to "GoogleCloud" from "GoogleCloudPowerShell". The
trailing "PowerShell" is redundant. (Although AWS' cmdlets are called
"AWSPowerShell".)

Changes the assumed location of the cmdlets.

Changes references to the product as a whole to "Cloud Tools for PowerShell", since "Google Cloud PowerShell" was the old name while we were still in alpha.

Previously the Cloud SDK installer
would the files in "${Cloud SDK}\platform\GoogleCloudPowerShell\" and add
"${Cloud SDK}\platform\GoogleCloudPowerShell\" to PSModulePath.

While this allowed PowerShell to quickly find GoogleCloudPowerShell.psd1,
it broke the ability to type Import-Module GoogleCloudPowerShell and do other
sorts of lookups.

Internal cl/133881461 will update the Cloud SDK to use a new folder for
PSModulePath: "${Cloud SDK}\platform\PowerShell". And the PowerShell
bits will be installed into "${Cloud SDK}\platform\PowerShell\GoogleCloud".

Testing:
- Manual testing that PowerShell behaves the way Microsoft tells us it does.
- Manually built the new installer, ran it, and manually replaced the "powershell"
    component bits ($INSTALLDIR/platform/) with the new bits.
- I did not test the PatchFromGcs.ps1 or AppendModulePath.ps1 scripts.

Fixes #314 and #315.

### CODE REVIEW

File: Google.PowerShell/Google.PowerShell.csproj [hunk 1]
```diff
@@ -195,9 +195,9 @@
       <SubType>Designer</SubType>
     </None>
     <None Include="README.md" />
-    <None Include="ReleaseFiles\BootstrapGoogleCloudPowerShell.ps1" />
+    <None Include="ReleaseFiles\BootstrapCloudToolsForPowerShell.ps1" />
     <None Include="ReleaseFiles\GoogleCloudPlatform.Format.ps1xml" />
-    <None Include="ReleaseFiles\GoogleCloudPowerShell.psd1" />
+    <None Include="ReleaseFiles\CloudToolsForPowerShell.psd1" />
```
Comment (ILMTitan) [inline Google.PowerShell/Google.PowerShell.csproj]: This should be GoogleCloud.psd1

Comment (chrsmith) [inline Google.PowerShell/Google.PowerShell.csproj]: Good catch. Evidently you need to "Save All" to save changes to the project file. (Which is something I'm sure I've been burned by in the past.)

**Comment by chrsmith** [comment]
PTAL

**Comment by ILMTitan** [comment]
LGTM

**Pull request merged**